gst/nnstreamer/tensor_data.c

```diff
@@ -0,0 +1,296 @@
+
+  td->data._int64_t = 0;
+  td->type = _NNS_END;
+
+  switch (type) {
+    case _NNS_INT32:
+      td_set_data (td, value, int32_t);
+      break;
+    case _NNS_UINT32:
+      td_set_data (td, value, uint32_t);
+      break;
+    case _NNS_INT16:
+      td_set_data (td, value, int16_t);
+      break;
+    case _NNS_UINT16:
+      td_set_data (td, value, uint16_t);
+      break;
+    case _NNS_INT8:
+      td_set_data (td, value, int8_t);
+      break;
+    case _NNS_UINT8:
+      td_set_data (td, value, uint8_t);
+      break;
+    case _NNS_FLOAT64:
+      td_set_data (td, value, double);
+      break;
+    case _NNS_FLOAT32:
+      td_set_data (td, value, float);
+      break;
+    case _NNS_INT64:
+      td_set_data (td, value, int64_t);
+      break;
+    case _NNS_UINT64:
+      td_set_data (td, value, uint64_t);
+      break;
+    default:
+      nns_logw ("Unknown tensor type %d", type);
+      return FALSE;
+  }
+
+  td->type = type;
+  return TRUE;
+}
+
+/**
+ * @brief Get tensor element value.
+ * @param td struct for tensor data
+ * @param value pointer of tensor element value
+ * @return TRUE if no error
+ */
+gboolean
+gst_tensor_data_get (tensor_data_s * td, gpointer value)
+{
+  g_return_val_if_fail (td != NULL, FALSE);
+  g_return_val_if_fail (value != NULL, FALSE);
+
+  switch (td->type) {
+    case _NNS_INT32:
+      td_get_data (td, value, int32_t);
+      break;
+    case _NNS_UINT32:
+      td_get_data (td, value, uint32_t);
+      break;
+    case _NNS_INT16:
+      td_get_data (td, value, int16_t);
+      break;
+    case _NNS_UINT16:
+      td_get_data (td, value, uint16_t);
+      break;
+    case _NNS_INT8:
+      td_get_data (td, value, int8_t);
+      break;
+    case _NNS_UINT8:
+      td_get_data (td, value, uint8_t);
+      break;
+    case _NNS_FLOAT64:
+      td_get_data (td, value, double);
+      break;
+    case _NNS_FLOAT32:
+      td_get_data (td, value, float);
+      break;
+    case _NNS_INT64:
+      td_get_data (td, value, int64_t);
+      break;
+    case _NNS_UINT64:
+      td_get_data (td, value, uint64_t);
+      break;
+    default:
+      nns_logw ("Unknown tensor type %d", td->type);
+      return FALSE;
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief Typecast tensor element data.
+ * @param td struct for tensor data
+ * @param type tensor type to be transformed
+ * @return TRUE if no error
+ */
+gboolean
+gst_tensor_data_typecast (tensor_data_s * td, tensor_type type)
+{
+  gboolean is_float;
+
+  g_return_val_if_fail (td != NULL, FALSE);
+  g_return_val_if_fail (type != _NNS_END, FALSE);
+
+  /* do nothing when transform to same type */
+  if (td->type != type) {
+    is_float = (td->type == _NNS_FLOAT32 || td->type == _NNS_FLOAT64);
+
+    switch (type) {
+      case _NNS_INT32:
+        td_typecast (td, int32_t);
+        break;
+      case _NNS_UINT32:
+        if (is_float) {
+          td_typecast (td, int32_t);
+          td->type = _NNS_INT32;
```

Comment:
Maybe this? line 233 overwrites the type
```suggestion
          type = _NNS_INT32;
```

Reply:
no, changing float > int32 > uint32 and float > uint32 is different.

Reply:
Ah I see :) Thank you for the explanation!